iib/workers/tasks/build.py

```diff
@@ -670,20 +677,23 @@ def handle_add_request(
 
     set_request_state(request_id, 'in_progress', 'Creating the manifest list')
     output_pull_spec = _create_and_push_manifest_list(request_id, arches)
-
-    if legacy_support_packages:
-        export_legacy_packages(
-            legacy_support_packages, request_id, output_pull_spec, cnr_token, organization
-        )
-
-    _finish_request_post_build(
+    modified_index_image = _update_index_image_pull_spec(
         output_pull_spec,
         request_id,
         arches,
         from_index,
         overwrite_from_index,
         overwrite_from_index_token,
     )
+    if legacy_support_packages:
+        export_legacy_packages(
+            legacy_support_packages, request_id, modified_index_image, cnr_token, organization
```

Comment:
I'd rather continue using `output_pull_spec` in this method since it will be using `quay.io` directly rather than the registry proxy, which is unreliable. The registry proxy is only provided as the `index_image` value in the REST API for convenience. IIB should use quay.io directly when possible.

Reply:
so that means if the index image is overridden by using overwrite_from_index, the push to OMPS should not consider it?

Reply:
No, I'm just saying that the push to OMPS should use the unaltered pull specification of the index image.